statistics/statistics.go

```diff
@@ -62,13 +65,15 @@ func New() *Plugin {
 	return p
 }
 
-func (p *Plugin) calculateStats(buff interface{}, logger *log.Logger) (map[string]float64, error) {
-	result := make(map[string]float64)
+// calculateStats calaculates the descriptive statistics for buff
+func (p *Plugin) calculateStats(buff interface{}, logger *log.Logger) (map[string][]float64, error) {
```

Comment:
don't pass log.Logger as the method argument. It does not need to be passed around. You can directly use it.
Such as log.Logger.Printf(...)